File: ruby/rotational-cipher/rotational_cipher.rb
class RotationalCipher
  UPPER = ('A'..'Z').to_a
  LOWER = ('a'..'z').to_a

  def self.rotate(text, steps)
    text.tr([UPPER, LOWER].join, [UPPER, LOWER].map { _1.rotate(steps) }.join)
  end
end

File: javascript/rotational-cipher/rotational-cipher.js
var RotationalCipher = function(){};

const CAPITAL = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'.split('');
const LOWERCASE = 'abcdefghijklmnopqrstuvwxyz'.split('');

RotationalCipher.prototype.rotate = function(phrase, distance){
  return phrase.split('').map(function(char){
    if (char.match(/[^A-Za-z]/)) {
      return char;
    }
    else {
      var set = char.match(/[A-Z]/) ? CAPITAL : LOWERCASE;
      return set[(set.indexOf(char) + distance) % 26];
    }
  }).join('');
};

module.exports = RotationalCipher;
